bin/cpu_monitor.py

```diff
@@ -91,7 +94,18 @@ def update_status_stale(stat, last_update_time):
 
 class CPUMonitor():
     def __init__(self, hostname, namespace, diag_hostname):
-        self._diag_pub = rospy.Publisher('/diagnostics', DiagnosticArray, queue_size = 100)
+        self._temp_diag_updater = DiagnosticUpdater(
+            '/ros_system_monitor/{}/cpu/temp'.format(namespace)
```

Comment:
I think I would just make one DiagnosticUpdater for the CPU. The temp and the usage can be their own individual GenericDiagnostic components that both feed into the one CPU status.

There's not really a need for an OutputDiagnostic, because this node doesn't publish anything. That's really designed to be used for monitoring publish frequencies and nothing depends on the specific timing of the CPU checks. If you just want to make sure the event loop that updates the CPU stats doesn't, for some reason stop working, you could do a really simple watchdog thread that just checks the last time it was updated or something. Much lighter weight than all the frequency monitoring logic wrapped up in the OutputDiagnostic. But even the watchdog thread I would say might be overkill, but up to you.

As a separate comment, we should try to standardize on ROS namespacing. We can actually just grab the namespace of this node directly from ROS and prepend it to the stat name. So it would be like `"{}/cpu".format(<get ros namespace>)`

These same comments apply to all the files in this PR.

Reply:
> I think I would just make one DiagnosticUpdater for the CPU. The temp and the usage can be their own individual GenericDiagnostic components that both feed into the one CPU status.

Done. This is what I initially had at first, but changed since RSP was sort of treating these as two different inputs, and since having two separate updaters makes it easier to distinguish which of the two root causes is creating a generic error message. But cool, happy to simplify it to a single updater for each monitor. Applied to both the CPU and HDD monitors.

> There's not really a need for an OutputDiagnostic

Done. Just removed it entirely. Our mbot diagnostics watchdog is watching this anyway, I just initially thought it was easy and nice to have. Yeah, I was re-using the machinery of the `OutputDiagnostic` just to easily monitor something that was operating kind of like a publisher, but good point that a simple watchdog thread would be much more light weight even if a bit more code here.

> As a separate comment, we should try to standardize on ROS namespacing

That's kinda what they're already doing in the main function at the bottom of this file already. That returns just `com-1` or `com-2`. Looks like in mbot we pass the namespace manually in the launch file around here: https://github.com/MarbleInc/mbot/blob/c90a2bf75bbd7509a427f4b8ffaed87cf38bf3b9/src/ros/mbot_bringup/drivers/drivers_burrata.launch#L86 . If we wanted to switch to just `{}/cpu.format(namespace)` here, we'd have to edit it in `drivers_burrata.launch` to something like `ns="system_monitor/com-1"` here (and maybe rename `timing_monitor.launch` to `ns="timing_monitor"`...). Thoughts?

Reply:
Cool, first two points look good.

For namespacing, I think the ros_system_monitor part of the namespace should also come from the launch file. You can have namespaces inside of namespaces. In the diagnostic, the namespace should always be prepended, not inserted into the middle. That means we'll probably end up with `/com-1/ros_system_monitor/cpu` which is a change but that's ok. It makes more sense that way and it's consistent with how the rest of ROS namespacing works.

Reply:
Gotcha, thanks for the notes on namespacing. Updated here in https://github.com/MarbleInc/ros-system-monitor/pull/3/commits/330f7b9ad0c8d7503ccfac420cc74e7a8b0fa662; corresponding mbot updates in https://github.com/MarbleInc/mbot/pull/3196/commits/2eb52b5cca9dffff98cb9b81b4c5c053108374bc as part of PR https://github.com/MarbleInc/mbot/pull/3196. More info in commit messages.

Hopefully this aligns well with what you were advising, but let me know if not, thanks!